### PR TITLE
2) solved task

### PR DESCRIPTION
**Pull request by AntonPanchishko**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -9,8 +9,89 @@
...
+                    <configuration>
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
 </project>
```
Comment (vicire) [inline pom.xml]: add empty line

File: src/main/java/config/AppConfig.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.orm.hibernate5.LocalSessionFactoryBean;
+
+@Configuration
+@ComponentScan(basePackages = {"service",
+        "dao"
+})
```
Comment (vicire) [inline src/main/java/config/AppConfig.java]:
```suggestion
@ComponentScan(basePackages = {"service", "dao"})
```

File: src/main/java/config/AppConfig.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
```
Comment (vicire) [inline src/main/java/config/AppConfig.java]: remove empty line

File: src/main/java/config/AppConfig.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+public class AppConfig {
+
```
Comment (vicire) [inline src/main/java/config/AppConfig.java]: Same here

File: pom.xml [hunk 5]
```diff
@@ -9,8 +9,89 @@
...
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.22</version>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (Resci) [inline pom.xml]: You use this dependency?

File: pom.xml (hunk 1 above)
Comment (Resci) [inline pom.xml]: Add new line at end of file

File: src/main/java/dao/impl/UserDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import dao.UserDao;
+import java.util.List;
+import model.User;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+    @Autowired
```
Comment (vicire) [inline src/main/java/dao/impl/UserDaoImpl.java:16]: What is better - field injection or constructor injection?

**Review by vicire — CHANGES_REQUESTED** [review]
Remove all redundant empty lines

File: src/main/java/dao/impl/UserDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    @Override
+    public void add(User user) {
+        sessionFactory.openSession().save(user);
+    }
+
+    @Override
+    public List<User> listUsers() {
+        return sessionFactory.openSession()
+                .createQuery("from User", User.class).getResultList();
+    }
```
Comment (Resci) [inline src/main/java/dao/impl/UserDaoImpl.java]: [Checklist](https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_01.html) P.5

**Review by Resci — APPROVED** [review]

**Review by Resci — APPROVED** [review]

File: src/main/java/service/impl/UserServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import service.UserService;
+
+@Service
+public class UserServiceImpl implements UserService {
+    private UserDao userDao;
+
+    @Autowired
```
Comment (dmytro-hryhoruk) [inline src/main/java/service/impl/UserServiceImpl.java:14]: @autowired annotation is not necessary above constructor. Let's either remove it from each constructor(with injection) or use it everywhere

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**